clouddriver-docker/src/test/java/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryServiceTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2025 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.docker.registry.api.v2.client;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import com.netflix.spinnaker.clouddriver.docker.registry.api.v2.auth.DockerBearerToken;
+import com.netflix.spinnaker.kork.retrofit.ErrorHandlingExecutorCallAdapterFactory;
+import com.netflix.spinnaker.kork.retrofit.Retrofit2SyncCall;
+import java.io.IOException;
+import java.time.Instant;
+import okhttp3.OkHttpClient;
+import okhttp3.ResponseBody;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.http.GET;
+import retrofit2.http.Header;
+import retrofit2.http.Headers;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public class DockerRegistryServiceTest {
+
+  static DockerRegistryClient.DockerRegistryService dockerRegistryService;
+  static TokenService tokenService;
+  String service = "registry.docker.io";
+  String scope = "repository:library/nginx:pull";
+  String repository = "library/nginx";
+  String tagsPath = "v2/library/nginx/tags/list";
+  private Instant expiryTime = Instant.now();
+  static String bearerToken;
+
+  @BeforeAll
+  public static void setup() {
+    tokenService = buildService(TokenService.class, "https://auth.docker.io");
+    dockerRegistryService =
+        buildService(
+            DockerRegistryClient.DockerRegistryService.class, "https://registry-1.docker.io");
+  }
+
+  @Test
+  void getToken() {
+    DockerBearerToken token =
+        Retrofit2SyncCall.execute(tokenService.getToken("token", service, scope, "spinnaker"));
```

Comment:
This makes actual calls to docker.io, yes?

Reply:
Yes. To quickly test the real docker registry apis using retrofit2 client.

Reply:
What do we lose by (wire)mocking this?  It doesn't seem OK to depend on real network calls like this.

Reply:
My intention was to demonstrate real calls only, though I agree we run a risk of external dependency. Other test class does use wirmock anyway, so if we don't want the real calls, I can delete this class.

Reply:
Yeah, fair point.  But if we can demonstrate the same bug in the code with a mock, yes please delete this file.

Reply:
The issue demonstration is happening in the other test so I just delete this file.